packages/teleport/src/Player/DesktopPlayer.tsx

```diff
@@ -25,39 +27,50 @@ export const DesktopPlayer = ({
     playerClient,
     tdpCliOnPngFrame,
     tdpCliOnClientScreenSpec,
-    showCanvas,
+    tdpCliOnWsClose,
+    tdpCliOnTdpError,
+    attempt,
   } = useDesktopPlayer({
     sid,
     clusterId,
   });
 
+  const displayCanvas = attempt.status === 'success' || attempt.status === '';
+  const displayProgressBar = attempt.status !== 'processing';
+
   return (
     <>
       <StyledPlayer>
-        {!showCanvas && (
+        {attempt.status === 'processing' && (
```

Comment:
Should you make constants or a type or something that defines the valid statuses so we don't just have magic strings scattered around?